nav2_collision_monitor/include/nav2_collision_monitor/source_base.hpp

```diff
@@ -0,0 +1,133 @@
+// Copyright (c) 2022 Samsung Research Russia
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef NAV2_COLLISION_MONITOR__SOURCE_BASE_HPP_
+#define NAV2_COLLISION_MONITOR__SOURCE_BASE_HPP_
+
+#include <vector>
+#include <string>
+
+#include "rclcpp/rclcpp.hpp"
+#include "nav2_util/lifecycle_node.hpp"
+
+#include "tf2/time.h"
+#include "tf2_ros/buffer.h"
+
+#include "nav2_collision_monitor/types.hpp"
+
+namespace nav2_collision_monitor
+{
+
+/**
+ * @brief Basic data source class
+ */
+class SourceBase
+{
+public:
+  /**
+   * @brief SourceBase constructor
+   */
+  SourceBase(
+    const nav2_util::LifecycleNode::WeakPtr & node,
+    std::shared_ptr<tf2_ros::Buffer> tf_buffer,
+    const std::string & source_name,
+    const std::string & base_frame_id,
+    const tf2::Duration & transform_tolerance,
+    const tf2::Duration & data_timeout);
+  /**
+   * @brief SourceBase destructor
+   */
+  virtual ~SourceBase();
+
+  /**
+   * @brief Data source configuration routine.
+   * Empty virtual method intended to be used in child implementations.
+   */
+  virtual void configure() = 0;
+
+  /**
+   * @brief Sets fixed frame ID. Used in data frame interpolation
+   * @param fixed_frame_id Fixed frame ID
+   */
+  void setFixedFrameId(const std::string & fixed_frame_id);
```

Comment:
`setGlobalFrameId()` Just to be consistent with other nodes

Reply:
There is no need to have this method in latest update